### PR TITLE
Building a Password Strength Classifier Model Using Machine Learning

### LINKED ISSUE #5722: [Machine Learning] Building a Password Strength Classifier Model Using Machine Learning
#### Proposed title of article
[Machine Learning] Building a Password Strength Classifier model using machine learning

#### Proposed article introduction
Password strength is a measure of the effectiveness of a password against guessing or brute-force attacks. In its usual form, it estimates how many trials an attacker who does not have direct access to the password would need, on average, to guess it correctly. The strength of a password is a function of length, complexity, and unpredictability.

Data breaches and identity theft are on the rise, and the cause is often compromised passwords. After stealing credentials, cybercriminals can use passwords to start disinformation campaigns against companies, use people’s payment information for purchases, and spy on users through WiFi-connected security cameras. Therefore, we need to build a model that can determine the strength of a given password, this enables us to have more secure systems that can not be easily cracked.

In this tutorial, we will build a password classification model using Scikit-Learn. The model will give the password from strength from the lowest strength to the highest strength depending on the password inputted by the user.

#### Key takeaways 
1. What are the impacts of having a week password.
2. How to generate secure passwords.
3. Dataset preprocessing.
4. Model building and training.
5. Using the model to classify the strength of a password.

#### Article quality
In this tutorial, we will build a machine model that we will be used to classify the strength of a given password. This ensures people use more secure passwords in order to prevent data breaches. We will implement this model from scratch so that any reader can easily follow the steps and come up with the model. In this tutorial, we will ensure the model makes accurate predictions. We thoroughly train the model to make sure it effectively learns.
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by kelvinkimani501**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5722

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by munubi254** [comment]
@hectorkambow, I'd like to review this article.

**Comment by Kayere** [comment]
Plagiarism check done ✅

File: content/articles/building-a-password-strength-classifier-model-using-machine-learning/index.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+```python
+len(allpasswords)
+```
+The output is shown below:
+
+```bash
+669639
+```
+From the image above this dataset is big and has many data samples, we can now use this dataset to build our model.
+
```
Comment (M0nicah) [inline content/articles/building-a-password-strength-classifier-model-using-machine-learning/index.md:215]: kindly add the image you are talking about

Comment (kelvinkimani501) [reply content/articles/building-a-password-strength-classifier-model-using-machine-learning/index.md:213]: Hello @M0nicah, that was an oversight. It's supposed to be from the `output` above not `image` above. I have made that change .
Thanks

**Comment by M0nicah** [comment]
🚀 awesome article @kelvinkimani501, please ensure you go through the changes I made in the commits to help improve flow and format for your future articles. Thanks.

**Review by M0nicah — APPROVED** [review]
🚀 awesome article @kelvinkimani501, please ensure you go through the changes I made in the commits to help improve flow and format for your future articles. Thanks.

**Comment by Kayere** [comment]
Awesome article @kelvinkimani501.
Thank you for the review @M0nicah.

**Pull request merged**